docs/developer-documentation.md

```diff
@@ -1,135 +1,179 @@
-## Anthias Developer Documentation
+# Developer documentation
 
-### Creating a new release
+## Understanding the components that make up Anthias
 
-Check what the latest release is:
+Here is a high-level overview of the different components that make Anthias:
 
-```bash
-$ git pull
-$ git tag
-[...]
-```
+![Anthias Diagram Overview](/docs/d2/anthias-diagram-overview.svg)
 
-Create a new release:
+These components and their dependencies are mostly installed and handled with Ansible and Docker.
+
+* The **NGINX** component (`anthias-nginx`) forwards requests to the backend and serves static files.
+* The **viewer** (`anthias-viewer`) is what drives the screen (e.g., shows web page, image or video).
+* The **web app** component (`anthias-server`) &mdash; which consists of the front-end and back-end code &ndash; is what the user interacts with via browser.
+* The **Celery** (`anthias-celery`) component is for aynschronouslt queueing and executing tasks outside the HTTP request-response cycle (e.g., doing assets cleanup).
+* The **WebSocket** (`anthias-websocket`) component is used for forwarding requests from NGINX to the backend.
+* **Redis** (`redis`) is used as a database, cache and message broker.
+* The **database** component uses **SQLite** for storing the assets information.
+
+There are currently three major branches in the repository:
+
+| Version       | Branch      | Comment    |
+| :------------ | :---------- | :--------- |
+|  Developer    | [`master`](https://github.com/Screenly/Anthias) | This is where we test things and apply latest fixes. |
+|  Production   | [`production`](https://github.com/Screenly/Anthias/tree/production) | This is the branch disk images are built from and should be properly tested |
+|  Experimental | [`experimental`](https://github.com/Screenly/Anthias/tree/experimental) | This is the branch for experimenting, such as using a new web browser |
```

Comment:
This needs to be overhauled. The `experimental` branch has been abandoned, and we might consider deprecating `production` too in order to move to rolling releases instead.